nsxt/resource_nsxt_vpc.go

```diff
@@ -0,0 +1,626 @@
+				Schema: map[string]*metadata.ExtendedSchema{
+					"enabled": {
+						Schema: schema.Schema{
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						Metadata: metadata.Metadata{
+							SchemaType:   "bool",
+							SdkFieldName: "Enabled",
+						},
+					},
+					"listener_ip": {
+						Schema: schema.Schema{
+							Type:         schema.TypeString,
+							ValidateFunc: validation.IsIPv4Address,
+							Optional:     true,
+						},
+						Metadata: metadata.Metadata{
+							SchemaType:   "string",
+							SdkFieldName: "ListenerIp",
+						},
+					},
+				},
+			},
+			Optional: true,
+		},
+		Metadata: metadata.Metadata{
+			SchemaType:   "struct",
+			SdkFieldName: "VpcDnsForwarder",
+			ReflectType:  reflect.TypeOf(model.VpcDnsForwarder{}),
+		},
+	},
+	"subnet_profiles": {
+		Schema: schema.Schema{
+			Type:     schema.TypeList,
+			MaxItems: 1,
+			Elem: &metadata.ExtendedResource{
+				Schema: map[string]*metadata.ExtendedSchema{
+					"ip_discovery": {
+						Schema: schema.Schema{
+							Type:         schema.TypeString,
+							ValidateFunc: validatePolicyPath(),
+							Optional:     true,
+						},
+						Metadata: metadata.Metadata{
+							SchemaType:   "string",
+							SdkFieldName: "IpDiscovery",
+						},
+					},
+					"qos": {
+						Schema: schema.Schema{
+							Type:         schema.TypeString,
+							ValidateFunc: validatePolicyPath(),
+							Optional:     true,
+						},
+						Metadata: metadata.Metadata{
+							SchemaType:   "string",
+							SdkFieldName: "Qos",
+						},
+					},
+					"segment_security": {
+						Schema: schema.Schema{
+							Type:         schema.TypeString,
+							ValidateFunc: validatePolicyPath(),
+							Optional:     true,
+						},
+						Metadata: metadata.Metadata{
+							SchemaType:   "string",
+							SdkFieldName: "SegmentSecurity",
+						},
+					},
+					"spoof_guard": {
+						Schema: schema.Schema{
+							Type:         schema.TypeString,
+							ValidateFunc: validatePolicyPath(),
+							Optional:     true,
+						},
+						Metadata: metadata.Metadata{
+							SchemaType:   "string",
+							SdkFieldName: "SpoofGuard",
+						},
+					},
+					"mac_discovery": {
+						Schema: schema.Schema{
+							Type:         schema.TypeString,
+							ValidateFunc: validatePolicyPath(),
+							Optional:     true,
+						},
+						Metadata: metadata.Metadata{
+							SchemaType:   "string",
+							SdkFieldName: "MacDiscovery",
+						},
+					},
+				},
+			},
+			Optional: true,
+		},
+		Metadata: metadata.Metadata{
+			SchemaType:   "struct",
+			SdkFieldName: "SubnetProfiles",
+			ReflectType:  reflect.TypeOf(model.SubnetProfiles{}),
+		},
+	},
+	"private_ipv4_blocks": {
+		Schema: schema.Schema{
+			Type: schema.TypeList,
+			Elem: &schema.Schema{
+				Type: schema.TypeString,
+			},
+			Optional: true,
+		},
+		Metadata: metadata.Metadata{
+			SchemaType:   "array",
+			SdkFieldName: "PrivateIpv4Blocks",
+		},
+	},
+	"short_id": {
```

Comment:
This is also computed as NSX sets value to it if user doesn't

Reply:
I wonder if this one is also impossible to update..

Reply:
NSX v4.1.2 UI protects it from updating. So I guess it's the same.